src/lib.rs

```diff
@@ -3199,6 +3176,44 @@ impl Connection {
                 }
 
                 self.do_handshake()?;
+
+                // Try to parse transport parameters as soon as the first flight
+                // of handshake data is processed.
+                //
+                // This is potentially dangerous as the handshake hasn't been
+                // completed yet, though it's required to be able to send data
+                // in 0.5 RTT.
+                let raw_params = self.handshake.quic_transport_params();
+
+                if !self.parsed_peer_transport_params && !raw_params.is_empty() {
```

Comment:
Note to self, will need to check that transport parameters have actually been sent by the client after handshake completion.

Reply:
Actually, we didn't check before, so it's not a regression.